// nba-scoreboard/build/constants/Constants.js
"use strict";

var BASE_NBA_DATA_URL = "http://data.nba.com/";
var BASE_NBA_DATA_SCOREBOARD_URL = BASE_NBA_DATA_URL.concat("data/5s/json/cms/noseason/scoreboard/");
var BASE_NBA_DATA_PLAY_BY_PLAY_URL = BASE_NBA_DATA_URL.concat("data/5s/json/cms/noseason/game/");

var DEFAULT_DATE_FORMAT = "YYYYMMDD";
var DEFAULT_TIMEZONE = "America/New_York";

var TRANSLATED_NBA_DATE_TIME_FORMAT = "YYYYMMDDHHmm";
var TRANSLATED_DATE_FORMAT = "LLL";

var PREGAME = 'PREGAME';
var LIVE = 'LIVE';
var FINAL = 'FINAL';

var TRANSLATED_GAME_STATUS_MAP = {
  1: PREGAME,
  2: LIVE,
  3: FINAL
};

var ONE_HUNDRED = 100;

var HOME_EMOJI_VALUE = 'house';
var VISITOR_EMOJI_VALUE = 'bus';
var START_TIME_EMOJI_VALUE = 'alarm_clock';
var BROADCASTS_EMOJI_VALUE = 'tv';
var SCORE_100_EMOJI_VALUE = '100';
var PLAY_TIME_EMOJI_VALUE = 'hourglass_flowing_sand';

var GAMES_OPTIONS = {
  TODAY: "TODAY",
  YESTERDAY: "YESTERDAY",
  TOMORROW: "TOMORROW"
};

module.exports = {
  BASE_NBA_DATA_SCOREBOARD_URL: BASE_NBA_DATA_SCOREBOARD_URL,
  BASE_NBA_DATA_PLAY_BY_PLAY_URL: BASE_NBA_DATA_PLAY_BY_PLAY_URL,
  DEFAULT_DATE_FORMAT: DEFAULT_DATE_FORMAT,
  DEFAULT_TIMEZONE: DEFAULT_TIMEZONE,
  TRANSLATED_NBA_DATE_TIME_FORMAT: TRANSLATED_NBA_DATE_TIME_FORMAT,
  TRANSLATED_DATE_FORMAT: TRANSLATED_DATE_FORMAT,
  PREGAME: PREGAME,
  LIVE: LIVE,
  FINAL: FINAL,
  TRANSLATED_GAME_STATUS_MAP: TRANSLATED_GAME_STATUS_MAP,
  HOME_EMOJI_VALUE: HOME_EMOJI_VALUE,
  VISITOR_EMOJI_VALUE: VISITOR_EMOJI_VALUE,
  START_TIME_EMOJI_VALUE: START_TIME_EMOJI_VALUE,
  BROADCASTS_EMOJI_VALUE: BROADCASTS_EMOJI_VALUE,
  SCORE_100_EMOJI_VALUE: SCORE_100_EMOJI_VALUE,
  PLAY_TIME_EMOJI_VALUE: PLAY_TIME_EMOJI_VALUE,
  GAMES_OPTIONS: GAMES_OPTIONS,
  ONE_HUNDRED: ONE_HUNDRED
};
