// nba-scoreboard/build/constants/TeamAbbreviations.js
'use strict';

var ATLANTA_HAWKS = 'ATL';
var BOSTON_CELTICS = 'BOS';
var CLEVELAND_CAVALIERS = 'CLE';
var NEW_ORLEANS_PELICANS = 'NOP';
var CHICAGO_BULLS = 'CHI';
var DALLAS_MAVERICKS = 'DAL';
var DENVER_NUGGETS = 'DEN';
var GOLDEN_STATE_WARRIORS = 'GSW';
var HOUSTON_ROCKETS = 'HOU';
var LOS_ANGELES_CLIPPERS = 'LAC';
var LOS_ANGELES_LAKERS = 'LAL';
var MIAMI_HEAT = 'MIA';
var MILWAUKEE_BUCKS = 'MIL';
var MINNESOTA_TIMBERWOLVES = 'MIN';
var BROOKLYN_NETS = 'BKN';
var NEW_YORK_KNICKS = 'NYK';
var ORLANDO_MAGIC = 'ORL';
var INDIANA_PACERS = 'IND';
var PHILADELPHIA_76ERS = 'PHI';
var PHOENIX_SUNS = 'PHX';
var PORTLAND_TRAIL_BLAZERS = 'POR';
var SACRAMENTO_KINGS = 'SAC';
var SAN_ANTONIO_SPURS = 'SAS';
var OKLAHOMA_CITY_THUNDER = 'OKC';
var TORONTO_RAPTORS = 'TOR';
var MEMPHIS_GRIZZLIES = 'MEM';
var WASHINGTON_WIZARDS = 'WAS';
var DETROIT_PISTONS = 'DET';
var CHARLOTTE_HORNETS = 'CHA';
var UTAH_JAZZ = 'UTA';

var ABBREVIATIONS = [ATLANTA_HAWKS, BOSTON_CELTICS, CLEVELAND_CAVALIERS, NEW_ORLEANS_PELICANS, CHICAGO_BULLS, DALLAS_MAVERICKS, DENVER_NUGGETS, GOLDEN_STATE_WARRIORS, HOUSTON_ROCKETS, LOS_ANGELES_CLIPPERS, LOS_ANGELES_LAKERS, MIAMI_HEAT, MILWAUKEE_BUCKS, MINNESOTA_TIMBERWOLVES, BROOKLYN_NETS, NEW_YORK_KNICKS, ORLANDO_MAGIC, INDIANA_PACERS, PHILADELPHIA_76ERS, PHOENIX_SUNS, PORTLAND_TRAIL_BLAZERS, SACRAMENTO_KINGS, SAN_ANTONIO_SPURS, OKLAHOMA_CITY_THUNDER, TORONTO_RAPTORS, MEMPHIS_GRIZZLIES, WASHINGTON_WIZARDS, DETROIT_PISTONS, CHARLOTTE_HORNETS, UTAH_JAZZ];

module.exports = {
  ATLANTA_HAWKS: ATLANTA_HAWKS,
  BOSTON_CELTICS: BOSTON_CELTICS,
  CLEVELAND_CAVALIERS: CLEVELAND_CAVALIERS,
  NEW_ORLEANS_PELICANS: NEW_ORLEANS_PELICANS,
  CHICAGO_BULLS: CHICAGO_BULLS,
  DALLAS_MAVERICKS: DALLAS_MAVERICKS,
  DENVER_NUGGETS: DENVER_NUGGETS,
  GOLDEN_STATE_WARRIORS: GOLDEN_STATE_WARRIORS,
  HOUSTON_ROCKETS: HOUSTON_ROCKETS,
  LOS_ANGELES_CLIPPERS: LOS_ANGELES_CLIPPERS,
  LOS_ANGELES_LAKERS: LOS_ANGELES_LAKERS,
  MIAMI_HEAT: MIAMI_HEAT,
  MILWAUKEE_BUCKS: MILWAUKEE_BUCKS,
  MINNESOTA_TIMBERWOLVES: MINNESOTA_TIMBERWOLVES,
  BROOKLYN_NETS: BROOKLYN_NETS,
  NEW_YORK_KNICKS: NEW_YORK_KNICKS,
  ORLANDO_MAGIC: ORLANDO_MAGIC,
  INDIANA_PACERS: INDIANA_PACERS,
  PHILADELPHIA_76ERS: PHILADELPHIA_76ERS,
  PHOENIX_SUNS: PHOENIX_SUNS,
  PORTLAND_TRAIL_BLAZERS: PORTLAND_TRAIL_BLAZERS,
  SACRAMENTO_KINGS: SACRAMENTO_KINGS,
  SAN_ANTONIO_SPURS: SAN_ANTONIO_SPURS,
  OKLAHOMA_CITY_THUNDER: OKLAHOMA_CITY_THUNDER,
  TORONTO_RAPTORS: TORONTO_RAPTORS,
  MEMPHIS_GRIZZLIES: MEMPHIS_GRIZZLIES,
  WASHINGTON_WIZARDS: WASHINGTON_WIZARDS,
  DETROIT_PISTONS: DETROIT_PISTONS,
  CHARLOTTE_HORNETS: CHARLOTTE_HORNETS,
  UTAH_JAZZ: UTAH_JAZZ,
  ABBREVIATIONS: ABBREVIATIONS
};
