// nba-scoreboard/build/services/tables/TableCreator.js
'use strict';
'use es6';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _cliTable = require('cli-table2');

var _cliTable2 = _interopRequireDefault(_cliTable);

var _immutable = require('immutable');

var _StartedGameTableCreator = require('./StartedGameTableCreator');

var _StartedGameTableCreator2 = _interopRequireDefault(_StartedGameTableCreator);

var _GamesTables = require('../../data/GamesTables');

var _GamesTables2 = _interopRequireDefault(_GamesTables);

var _PlaysTableCreator = require('./PlaysTableCreator');

var _PlaysTableCreator2 = _interopRequireDefault(_PlaysTableCreator);

var _TeamBoxScoreLeadersTableCreator = require('./TeamBoxScoreLeadersTableCreator');

var _TeamBoxScoreLeadersTableCreator2 = _interopRequireDefault(_TeamBoxScoreLeadersTableCreator);

var _UpcomingGamesTableCreator = require('./UpcomingGamesTableCreator');

var _UpcomingGamesTableCreator2 = _interopRequireDefault(_UpcomingGamesTableCreator);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var TableCreator = function () {
  function TableCreator() {
    _classCallCheck(this, TableCreator);
  }

  _createClass(TableCreator, null, [{
    key: 'create',
    value: function create(data) {
      var upcomingTable = data.upcoming.isEmpty() ? undefined : _UpcomingGamesTableCreator2.default.create(data.upcoming);
      return new _GamesTables2.default({
        started: (0, _immutable.List)(data.started.map(function (game) {
          return TableCreator.createGameTable(game);
        })),
        upcoming: upcomingTable
      });
    }
  }, {
    key: 'createGameTable',
    value: function createGameTable(game) {
      var table = new _cliTable2.default();
      table.push(TableCreator.createStartedeGameTable(game.metadata).toJS());
      table.push(TableCreator.createBoxScoreLeadersTables(game.boxScoreLeaders).toJS());
      table.push(TableCreator.createPlaysTable(game.plays).toJS());
      return table.toString();
    }
  }, {
    key: 'createStartedeGameTable',
    value: function createStartedeGameTable(data) {
      return _immutable.List.of((0, _immutable.Map)({
        content: _StartedGameTableCreator2.default.create(data),
        colSpan: 2,
        hAlign: 'center'
      }));
    }
  }, {
    key: 'createBoxScoreLeadersTables',
    value: function createBoxScoreLeadersTables(boxScoreLeaders) {
      return _immutable.List.of((0, _immutable.Map)({
        content: _TeamBoxScoreLeadersTableCreator2.default.create(boxScoreLeaders.home, true),
        colSpan: 1,
        hAlign: 'center'
      }), (0, _immutable.Map)({
        content: _TeamBoxScoreLeadersTableCreator2.default.create(boxScoreLeaders.visitor, false),
        colSpan: 1,
        hAlign: 'center'
      }));
    }
  }, {
    key: 'createPlaysTable',
    value: function createPlaysTable(plays) {
      return _immutable.List.of((0, _immutable.Map)({
        content: _PlaysTableCreator2.default.create(plays),
        colSpan: 2,
        hAlign: 'center'
      }));
    }
  }]);

  return TableCreator;
}();

exports.default = TableCreator;
