// nba-scoreboard/build/services/tables/TeamBoxScoreLeadersTableCreator.js
'use strict';
'use es6';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _colors = require('colors');

var _colors2 = _interopRequireDefault(_colors);

var _nodeEmoji = require('node-emoji');

var _nodeEmoji2 = _interopRequireDefault(_nodeEmoji);

var _cliTable = require('cli-table2');

var _cliTable2 = _interopRequireDefault(_cliTable);

var _immutable = require('immutable');

var _Constants = require('../../constants/Constants');

var _Constants2 = _interopRequireDefault(_Constants);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var TeamBoxScoreLeadersTableCreator = function () {
  function TeamBoxScoreLeadersTableCreator() {
    _classCallCheck(this, TeamBoxScoreLeadersTableCreator);
  }

  _createClass(TeamBoxScoreLeadersTableCreator, null, [{
    key: 'create',
    value: function create(data, isHome) {
      var table = TeamBoxScoreLeadersTableCreator.buildInitialTable(isHome);
      TeamBoxScoreLeadersTableCreator.buildRows(data).forEach(function (row) {
        return table.push(row.toJS());
      });
      return table.toString();
    }
  }, {
    key: 'buildInitialTable',
    value: function buildInitialTable(isHome) {
      var value = isHome ? 'Home' : 'Away';

      return new _cliTable2.default({
        head: [{
          content: value + ' Leaders',
          colSpan: 3,
          hAlign: 'center'
        }]
      });
    }
  }, {
    key: 'buildRows',
    value: function buildRows(data) {
      var rows = (0, _immutable.List)();
      rows = rows.push(_immutable.List.of('Points', data.points.value, data.points.getLeadersAbbreviatedNames()));
      rows = rows.push(_immutable.List.of('Assists', data.assists.value, data.assists.getLeadersAbbreviatedNames()));
      rows = rows.push(_immutable.List.of('Rebounds', data.rebounds.value, data.rebounds.getLeadersAbbreviatedNames()));
      return rows;
    }
  }]);

  return TeamBoxScoreLeadersTableCreator;
}();

exports.default = TeamBoxScoreLeadersTableCreator;
