// nba-scoreboard/build/services/translators/BoxScoreDataTranslator.js
'use strict';
'use es6';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _immutable = require('immutable');

var _GameBoxScoreLeaders = require('../../data/GameBoxScoreLeaders');

var _GameBoxScoreLeaders2 = _interopRequireDefault(_GameBoxScoreLeaders);

var _TeamBoxScoreLeaders = require('../../data/TeamBoxScoreLeaders');

var _TeamBoxScoreLeaders2 = _interopRequireDefault(_TeamBoxScoreLeaders);

var _StatisticalLeaders = require('../../data/StatisticalLeaders');

var _StatisticalLeaders2 = _interopRequireDefault(_StatisticalLeaders);

var _Player = require('../../data/Player');

var _Player2 = _interopRequireDefault(_Player);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var BoxScoreDataTranslator = function () {
  function BoxScoreDataTranslator() {
    _classCallCheck(this, BoxScoreDataTranslator);
  }

  _createClass(BoxScoreDataTranslator, null, [{
    key: 'translateStatLeaders',
    value: function translateStatLeaders(data) {
      var leaders = (0, _immutable.List)(data.leader.map(function (leader) {
        return new _Player2.default({
          firstName: leader.FirstName,
          lastName: leader.LastName
        });
      }));
      return new _StatisticalLeaders2.default({ value: parseInt(data.StatValue), leaders: leaders });
    }
  }, {
    key: 'translateBoxScoreData',
    value: function translateBoxScoreData(data) {
      var visitorLeaders = data.sports_content.game.visitor.Leaders;
      var homeLeaders = data.sports_content.game.home.Leaders;

      var visitorPointsLeaders = new _StatisticalLeaders2.default();
      if (visitorLeaders.hasOwnProperty('Points')) {
        visitorPointsLeaders = BoxScoreDataTranslator.translateStatLeaders(visitorLeaders.Points);
      }

      var visitorAssistsLeaders = new _StatisticalLeaders2.default();
      if (visitorLeaders.hasOwnProperty('Assists')) {
        visitorAssistsLeaders = BoxScoreDataTranslator.translateStatLeaders(visitorLeaders.Assists);
      }

      var visitorReboundsLeaders = new _StatisticalLeaders2.default();
      if (visitorLeaders.hasOwnProperty('Rebounds')) {
        visitorReboundsLeaders = BoxScoreDataTranslator.translateStatLeaders(visitorLeaders.Rebounds);
      }

      var homePointsLeaders = new _StatisticalLeaders2.default();
      if (homeLeaders.hasOwnProperty('Points')) {
        homePointsLeaders = BoxScoreDataTranslator.translateStatLeaders(homeLeaders.Points);
      }

      var homeAssistsLeaders = new _StatisticalLeaders2.default();
      if (homeLeaders.hasOwnProperty('Assists')) {
        homeAssistsLeaders = BoxScoreDataTranslator.translateStatLeaders(homeLeaders.Assists);
      }

      var homeReboundsLeaders = new _StatisticalLeaders2.default();
      if (homeLeaders.hasOwnProperty('Rebounds')) {
        homeReboundsLeaders = BoxScoreDataTranslator.translateStatLeaders(homeLeaders.Rebounds);
      }

      var visitorBoxScoreLeaders = new _TeamBoxScoreLeaders2.default({
        points: visitorPointsLeaders,
        assists: visitorAssistsLeaders,
        rebounds: visitorReboundsLeaders
      });

      var homeBoxScoreLeaders = new _TeamBoxScoreLeaders2.default({
        points: homePointsLeaders,
        assists: homeAssistsLeaders,
        rebounds: homeReboundsLeaders
      });
      return new _GameBoxScoreLeaders2.default({ home: homeBoxScoreLeaders, visitor: visitorBoxScoreLeaders });
    }
  }]);

  return BoxScoreDataTranslator;
}();

exports.default = BoxScoreDataTranslator;
