// nba-scoreboard/build/services/translators/PlaysTranslator.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _immutable = require('immutable');

var _Play = require('../../data/Play');

var _Play2 = _interopRequireDefault(_Play);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var PlaysTranslator = function () {
  function PlaysTranslator() {
    _classCallCheck(this, PlaysTranslator);
  }

  _createClass(PlaysTranslator, null, [{
    key: 'translate',
    value: function translate(data) {
      var plays = data.sports_content.game.play;
      var index = Math.min(5, plays.length);
      var recentPlays = plays.slice(-index);
      return (0, _immutable.List)(recentPlays.map(function (play) {
        return PlaysTranslator.buildPlay(play);
      }));
    }
  }, {
    key: 'buildPlay',
    value: function buildPlay(play) {
      return new _Play2.default({
        description: play.description,
        clock: play.clock,
        period: parseInt(play.period),
        teamAbbreviation: play.team_abr
      });
    }
  }]);

  return PlaysTranslator;
}();

exports.default = PlaysTranslator;
