// nba-scoreboard/build/services/translators/PlayByPlayTranslator.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _immutable = require('immutable');

var _PlayByPlay = require('../../data/PlayByPlay');

var _PlayByPlay2 = _interopRequireDefault(_PlayByPlay);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var PlayByPlayTranslator = function () {
  function PlayByPlayTranslator() {
    _classCallCheck(this, PlayByPlayTranslator);
  }

  _createClass(PlayByPlayTranslator, null, [{
    key: 'translate',
    value: function translate(data) {
      var playByPlayData = data.sports_content.game.play;
      var index = Math.min(5, playByPlayData.length);
      var recentPlays = playByPlayData.slice(-index);
      return (0, _immutable.List)(recentPlays.map(function (play) {
        return PlayByPlayTranslator.buildPlayByPlay(play);
      }));
    }
  }, {
    key: 'buildPlayByPlay',
    value: function buildPlayByPlay(play) {
      return new _PlayByPlay2.default({
        description: play.description,
        clock: play.clock,
        period: parseInt(play.period),
        teamAbbreviation: play.team_abr
      });
    }
  }]);

  return PlayByPlayTranslator;
}();

exports.default = PlayByPlayTranslator;
