// nba-scoreboard/build/services/translators/ScoreboardGameTranslator.js
'use strict';
'use es6';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _immutable = require('immutable');

var _momentTimezone = require('moment-timezone');

var _momentTimezone2 = _interopRequireDefault(_momentTimezone);

var _jstimezonedetect = require('jstimezonedetect');

var _jstimezonedetect2 = _interopRequireDefault(_jstimezonedetect);

var _GameStatus = require('../../data/GameStatus');

var _GameStatus2 = _interopRequireDefault(_GameStatus);

var _GameScoring = require('../../data/GameScoring');

var _GameScoring2 = _interopRequireDefault(_GameScoring);

var _GameScoreboard = require('../../data/GameScoreboard');

var _GameScoreboard2 = _interopRequireDefault(_GameScoreboard);

var _Constants = require('../../constants/Constants');

var _Constants2 = _interopRequireDefault(_Constants);

var _Location = require('../../data/Location');

var _Location2 = _interopRequireDefault(_Location);

var _Period = require('../../data/Period');

var _Period2 = _interopRequireDefault(_Period);

var _PeriodScore = require('../../data/PeriodScore');

var _PeriodScore2 = _interopRequireDefault(_PeriodScore);

var _Score = require('../../data/Score');

var _Score2 = _interopRequireDefault(_Score);

var _Matchup = require('../../data/Matchup');

var _Matchup2 = _interopRequireDefault(_Matchup);

var _Team = require('../../data/Team');

var _Team2 = _interopRequireDefault(_Team);

var _Broadcast = require('../../data/Broadcast');

var _Broadcast2 = _interopRequireDefault(_Broadcast);

var _BroadcastMedium = require('../../data/BroadcastMedium');

var _BroadcastMedium2 = _interopRequireDefault(_BroadcastMedium);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var ScoreboardGameTranslator = function () {
  function ScoreboardGameTranslator() {
    _classCallCheck(this, ScoreboardGameTranslator);
  }

  _createClass(ScoreboardGameTranslator, null, [{
    key: 'translate',
    value: function translate(data) {

      if (!('period_time' in data)) {
        throw new ReferenceError('period_time field missing');
      }

      if (!('broadcasters' in data)) {
        throw new ReferenceError('broadcasters field missing');
      }

      if (!('home' in data)) {
        throw new ReferenceError('home field missing');
      }

      if (!('visitor' in data)) {
        throw new ReferenceError('visitor field missing');
      }

      var periodTime = data.period_time;
      var broadcasters = data.broadcasters;
      var homeData = data.home;
      var awayData = data.visitor;

      return new _GameScoreboard2.default({
        id: data.id,
        status: ScoreboardGameTranslator.getGameStatus(periodTime),
        startTimestamp: ScoreboardGameTranslator.getStartTimestamp(data),
        location: ScoreboardGameTranslator.getLocation(data),
        period: ScoreboardGameTranslator.getPeriod(periodTime),
        broadcasts: ScoreboardGameTranslator.getBroadcasts(broadcasters),
        matchup: ScoreboardGameTranslator.getMatchup(homeData, awayData),
        scoring: ScoreboardGameTranslator.getScoring(homeData, awayData)
      });
    }
  }, {
    key: 'getGameStatus',
    value: function getGameStatus(periodTime) {
      if (!'game_status' in periodTime) {
        throw new ReferenceError('game_status field not in data');
      }

      var _iteratorNormalCompletion = true;
      var _didIteratorError = false;
      var _iteratorError = undefined;

      try {
        for (var _iterator = _GameStatus2.default.enumValues[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
          var status = _step.value;

          if (status.nbaStatsGameStatus == periodTime.game_status) {
            return status;
          }
        }
      } catch (err) {
        _didIteratorError = true;
        _iteratorError = err;
      } finally {
        try {
          if (!_iteratorNormalCompletion && _iterator.return) {
            _iterator.return();
          }
        } finally {
          if (_didIteratorError) {
            throw _iteratorError;
          }
        }
      }

      throw new ReferenceError('unknown nba stats game status');
    }
  }, {
    key: 'getStartTimestamp',
    value: function getStartTimestamp(gameData) {
      if (!('date' in gameData)) {
        throw new ReferenceError('date field missing');
      }

      if (!('time' in gameData)) {
        throw new ReferenceError('time field missing');
      }

      var rawStartTime = '' + gameData.date + gameData.time;

      return _momentTimezone2.default.tz(rawStartTime, _Constants2.default.TRANSLATED_NBA_DATE_TIME_FORMAT, _Constants2.default.DEFAULT_TIMEZONE).clone().tz("UTC");
    }
  }, {
    key: 'getLocation',
    value: function getLocation(gameData) {
      if (!('arena' in gameData)) {
        throw new ReferenceError('arena field missing');
      }

      if (!('city' in gameData)) {
        throw new ReferenceError('city field missing');
      }

      if (!('state' in gameData)) {
        throw new ReferenceError('state field missing');
      }

      return new _Location2.default({
        arena: gameData.arena,
        city: gameData.city,
        state: gameData.state
      });
    }
  }, {
    key: 'getPeriod',
    value: function getPeriod(periodTime) {
      if (!('period_status' in periodTime)) {
        throw new ReferenceError('period_status field missing');
      }

      if (!('period_value' in periodTime)) {
        throw new ReferenceError('period_value field missing');
      }

      if (!('game_clock' in periodTime)) {
        throw new ReferenceError('game_clock field missing');
      }

      return new _Period2.default({
        value: parseInt(periodTime.period_value),
        status: periodTime.period_status,
        clock: periodTime.game_clock
      });
    }
  }, {
    key: 'getBroadcasts',
    value: function getBroadcasts(broadcasters) {
      if (!('radio' in broadcasters)) {
        throw new ReferenceError('radio field missing');
      }

      if (!('tv' in broadcasters)) {
        throw new ReferenceError('tv field missing');
      }

      if (!('broadcaster' in broadcasters.radio)) {
        throw new ReferenceError('radio broadcasters field missing');
      }

      if (!('broadcaster' in broadcasters.tv)) {
        throw new ReferenceError('tv broadcasters field missing');
      }

      var radioBroadcasters = (0, _immutable.List)(broadcasters.radio.broadcaster.map(function (broadcast) {
        return ScoreboardGameTranslator.getBroadcast(broadcast, _BroadcastMedium2.default.RADIO);
      }));
      var tvBroadcasters = (0, _immutable.List)(broadcasters.tv.broadcaster.map(function (broadcast) {
        return ScoreboardGameTranslator.getBroadcast(broadcast, _BroadcastMedium2.default.TV);
      }));

      return tvBroadcasters.concat(radioBroadcasters);
    }
  }, {
    key: 'getBroadcast',
    value: function getBroadcast(broadcast, medium) {
      if (!('scope' in broadcast)) {
        throw new ReferenceError('broadcast scope field missing');
      }

      if (!('display_name' in broadcast)) {
        throw new ReferenceError('broadcast display name field missing');
      }

      if (!(medium instanceof _BroadcastMedium2.default)) {
        throw new TypeError('medium must be a BroadcastMedium');
      }

      return new _Broadcast2.default({
        scope: broadcast.scope,
        name: broadcast.display_name,
        medium: medium
      });
    }
  }, {
    key: 'getMatchup',
    value: function getMatchup(homeData, awayData) {
      return new _Matchup2.default({
        homeTeam: ScoreboardGameTranslator.getTeam(homeData),
        awayTeam: ScoreboardGameTranslator.getTeam(awayData)
      });
    }
  }, {
    key: 'getTeam',
    value: function getTeam(team) {
      if (!('city' in team)) {
        throw new ReferenceError('city field missing');
      }

      if (!('nickname' in team)) {
        throw new ReferenceError('nickname field missing');
      }

      if (!('abbreviation' in team)) {
        throw new ReferenceError('abbreviation field missing');
      }

      return new _Team2.default({
        city: team.city,
        nickname: team.nickname,
        abbreviation: team.abbreviation
      });
    }
  }, {
    key: 'getScoring',
    value: function getScoring(homeData, awayData) {
      return new _GameScoring2.default({
        periods: ScoreboardGameTranslator.getPeriodScores(homeData, awayData),
        total: ScoreboardGameTranslator.getTotalScore(homeData, awayData)
      });
    }
  }, {
    key: 'getPeriodScores',
    value: function getPeriodScores(homeData, awayData) {
      // if period scores are not found, return an empty list
      if (!('linescores' in homeData)) {
        return (0, _immutable.List)();
      }

      if (!('linescores' in awayData)) {
        return (0, _immutable.List)();
      }

      if (!('period' in homeData.linescores)) {
        return (0, _immutable.List)();
      }

      if (!('period' in awayData.linescores)) {
        return (0, _immutable.List)();
      }

      var homePeriodScores = homeData.linescores.period;
      var awayPeriodScores = awayData.linescores.period;

      if (homePeriodScores.length != awayPeriodScores.length) {
        throw new Error('home period scores length is not equal to the away period scores length');
      }

      var periodScores = (0, _immutable.List)();
      for (var index = 0; index < homePeriodScores.length; index++) {
        var homePeriodScore = homePeriodScores[index];
        var awayPeriodScore = awayPeriodScores[index];
        periodScores = periodScores.push(ScoreboardGameTranslator.getPeriodScore(homePeriodScore, awayPeriodScore));
      }

      return periodScores;
    }
  }, {
    key: 'getPeriodScore',
    value: function getPeriodScore(homePeriodScore, awayPeriodScore) {
      if (!('period_value' in homePeriodScore)) {
        throw new ReferenceError('home period value field missing');
      }

      if (!('score' in homePeriodScore)) {
        throw new ReferenceError('home score field missing');
      }

      if (!('period_value' in awayPeriodScore)) {
        throw new ReferenceError('away period value field missing');
      }

      if (!('score' in awayPeriodScore)) {
        throw new ReferenceError('away score field missing');
      }

      if (homePeriodScore.period_value != awayPeriodScore.period_value) {
        throw new ReferenceError('different period values');
      }

      return new _PeriodScore2.default({
        period: parseInt(homePeriodScore.period_value),
        score: new _Score2.default({
          home: parseInt(homePeriodScore.score),
          away: parseInt(awayPeriodScore.score)
        })
      });
    }
  }, {
    key: 'getTotalScore',
    value: function getTotalScore(homeData, awayData) {
      if (!('score' in homeData)) {
        return new _Score2.default();
      }

      if (!('score' in awayData)) {
        return new _Score2.default();
      }

      return new _Score2.default({
        home: parseInt(homeData.score),
        away: parseInt(awayData.score)
      });
    }
  }]);

  return ScoreboardGameTranslator;
}();

exports.default = ScoreboardGameTranslator;
