// nba-scoreboard/build/tables/UpcomingGamesTableCreator.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _cliTable = require('cli-table2');

var _cliTable2 = _interopRequireDefault(_cliTable);

var _immutable = require('immutable');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var UpcomingGamesTableCreator = function () {
  function UpcomingGamesTableCreator() {
    _classCallCheck(this, UpcomingGamesTableCreator);
  }

  _createClass(UpcomingGamesTableCreator, null, [{
    key: 'create',
    value: function create(data) {
      var table = new _cliTable2.default(UpcomingGamesTableCreator.getTableConfiguration().toJS());
      data.forEach(function (metadata) {
        return table.push(UpcomingGamesTableCreator.formatMetadata(metadata).toJS());
      });
      return table.toString();
    }
  }, {
    key: 'formatMetadata',
    value: function formatMetadata(metadata) {
      return _immutable.List.of(metadata.getLocalizedStartDateTime(), metadata.home.getName(), metadata.visitor.getName(), metadata.getBroadcastsString(), metadata.location.getFormattedLocation());
    }
  }, {
    key: 'getTableConfiguration',
    value: function getTableConfiguration() {
      return (0, _immutable.Map)({
        head: UpcomingGamesTableCreator.getHeader()
      });
    }
  }, {
    key: 'getHeader',
    value: function getHeader() {
      return _immutable.List.of('UPCOMING', 'HOME', 'AWAY', 'WATCH', 'ARENA');
    }
  }]);

  return UpcomingGamesTableCreator;
}();

exports.default = UpcomingGamesTableCreator;
